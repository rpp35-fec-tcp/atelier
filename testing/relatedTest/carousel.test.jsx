/* eslint-disable no-undef */
/* eslint-disable react/jsx-no-undef */
import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import renderer from 'react-test-renderer';
import SimpleCarousel from '../../client/src/components/relatedItems&comparison/carousel';
import fakeProductInfos from './fakeDataInfo';

describe('test SimpleCarousal', () => {
  test('snapshot test', () => {
    const modal = renderer.create(<SimpleCarousel
      relatedProducts={[71699, 71698, 71704, 71703]}
      currentProductId="71704"
      currentProductInfo={fakeProductInfos[0]}
    />).toJSON();
    expect(modal).toMatchSnapshot();
  });
  test('snapshot test', () => {
    const modal = renderer.create(<SimpleCarousel
      relatedProducts={[null, 71699, 71698, 71704, 71703]}
      currentProductId="71704"
      currentProductInfo={fakeProductInfos[0]}
    />).toJSON();
    expect(modal).toMatchSnapshot();
  });
});
