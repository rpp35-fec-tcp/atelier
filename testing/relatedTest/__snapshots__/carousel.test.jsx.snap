// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test SimpleCarousal snapshot test 1`] = `
<div>
  <div
    className="react-multi-carousel-list  "
  >
    <ul
      className="react-multi-carousel-track "
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseMove={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "overflow": "unset",
          "transform": "translate3d(0px,0,0)",
          "transition": "none",
        }
      }
    />
  </div>
</div>
`;

exports[`test SimpleCarousal snapshot test 2`] = `
<div>
  <div
    className="react-multi-carousel-list  "
  >
    <ul
      className="react-multi-carousel-track "
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseMove={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "overflow": "unset",
          "transform": "translate3d(0px,0,0)",
          "transition": "none",
        }
      }
    />
  </div>
</div>
`;
